src/interceptors/transform-response.interceptor.service.ts

```diff
@@ -0,0 +1,15 @@
+import { type CallHandler, type ExecutionContext, Injectable, type NestInterceptor } from '@nestjs/common';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+@Injectable()
+export class TranformResponseInterceptor<T> implements NestInterceptor<T, { status: boolean; data: T }> {
+  intercept(_context: ExecutionContext, next: CallHandler): Observable<{ status: boolean; data: T }> {
+    return next.handle().pipe(
+      map((data) => ({
+        status: !!data,
+        data
+      })),
```

Comment:
the response data not enough 

Ref:

```
{
  "status": bool,
  "message": string,
  "data": array | object | bool
  "metadata": any
}
```
Sometime meta data contains pagination data like page,  total ...